app/controllers/admin/deals_controller.rb

```diff
@@ -0,0 +1,55 @@
+class Admin::DealsController < Admin::BaseController
+  before_action :requested_deal, only: [:show, :edit, :destroy, :update]
+  
+  def index
+    @deals = Deal.where(status: params[:status])
+    @status = params[:status]
+  end
+
+  def new
+    @deal = Deal.new
+  end
+
+  def create
+    deal = Deal.new(deal_permitted_params)
+    deal.admin = logged_in_user
+
+    if deal.save
+      redirect_to admin_deals_url(status: 'unpublished')
+    else
+      redirect_to new_admin_deal_url, notice: t(:default_error_message)
```

Comment:
let's render :new so we can show inline error messages

Reply:
Implemented inline errors, handling else block by rendering instead of redirection